pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -299,12 +301,15 @@ public PersistentTopic(String topic, ManagedLedger ledger, BrokerService brokerS
         TopicName topicName = TopicName.get(topic);
         if (brokerService.getPulsar().getConfiguration().isTransactionCoordinatorEnabled()
                 && !isEventSystemTopic(topicName)) {
+            this.transactionEnabled = false;
             this.transactionBuffer = brokerService.getPulsar()
                     .getTransactionBufferProvider().newTransactionBuffer(this);
+            this.transactionBuffer.syncMaxReadPositionForNormalPublish((PositionImpl) ledger.getLastConfirmedEntry());
         } else {
+            this.transactionEnabled = true;
             this.transactionBuffer = new TransactionBufferDisable();
```

Comment:
why not use TransactionBufferDisable directly? in this way, we don't need check the transactionEnabled